src/renderer/atlas/shader_ps.hlsl

```diff
@@ -30,6 +33,11 @@ Output main(PSData data) : SV_Target
     float4 color;
     float4 weights;
 
+    // When drawing (dotted/dashed/curly) lines, texcoord holds the line
+    // rendention scale (1x or 2x) which we use to draw wide/tall lines.
+    const uint horizontalScale = data.texcoord.x;
+    const uint verticalScale = data.texcoord.y;

```

Comment:
I just realized an alternative idea I've had before... `shadingType` is a `u32` due to alignment requirements and we only use the lowest 3 bits right now. We could split it into two 16-bit values and use
```cpp
switch (data.shadingType & 0xffff)
```
here and extract the horizontal scale with `data.shadingType & 0x10000`, etc., in the branches below. That might simplify some of the code on the CPU side too, because we then don't need to carefully handle this in `BackendD3D::_drawCursorForegroundSlowPath` or in the vertex shader. Just an idea though...

Reply:
> shadingType is a u32 due to alignment requirements and we only use the lowest 3 bits right now. We could split it into two 16-bit values

This is a really good idea. I followed mostly what you mentioned, but instead of bit manipulation, I modified the `QuadInstance` to carry the rendition scale as a separate member. This is done by shrinking `ShadingType` to `u16` (from `u32`), and adding `renditionScale` of `u8x2` type.

This simplified the code in two folds:
- No special handling required for `texcoord`.
- Access pattern for `AtlasGlyphEntryData::shadingType` is made simpler. No need to cast `ShadingType` to `u16` anymore.